### Data Analysis with Panda
- Bike Sales
- Michael Hendricks
- July 20

In [135]:
import pandas as pd
import numpy as np

In [136]:
station_df = pd.read_csv('data/station.csv')
trip_df = pd.read_csv('data/trip.csv', error_bad_lines=False)
weather_df = pd.read_csv('data/weather.csv')

b'Skipping line 50794: expected 12 fields, saw 20\n'


In [156]:
trip_df[['tripduration']].mean()

tripduration    1178.295675
dtype: float64

### 1. The average trip duration for a borrowed bicycle is 1178.295675

In [157]:
trip_df['birthyear'].mode()

0    1987.0
dtype: float64

### 2. The most common birth year of a bicycle sharer is 1987, and since that's the year I was born, that makes them 33

In [158]:
#weather_df['Date'] = pd.to_datetime(weather_df['Date'], format='%m/%d/%Y')
#weather_df.loc[weather_df['Date'].dt.month == 12][['Date', 'Precipitation_In']].mean()

In [159]:
prec_per_day = weather_df[['Date','Precipitation_In']].mean()

In [160]:
prec_per_month = prec_per_day * 30

In [161]:
prec_per_month

Precipitation_In    3.151959
dtype: float64

### 3. The average precipitation per month is 3.15

In [165]:
#station_df[['name','current_dockcount']]
station_df[['current_dockcount']].mean()

current_dockcount    16.517241
dtype: float64

### 4. The average number of bikes at a given station is 16.517241

In [166]:
# notnull() filters out NaN returns
modified_stations = station_df[station_df['modification_date'].notnull()][['name','install_dockcount', 'modification_date', 'current_dockcount']].set_index('name')

In [168]:
modified_stations[['install_dockcount']].mean()

install_dockcount    17.529412
dtype: float64

In [169]:
modified_stations[['current_dockcount']].mean()

current_dockcount    13.764706
dtype: float64

### 5. When a bike station is modified it is more likely that it's lose bikes because the average dock count of stations before modifications was higher than the average dock count of those same stations after modifications

In [184]:
weather_df[['Max_Temperature_F']].max()

Max_Temperature_F    98
dtype: int64

### 6. The hottest day on record hit a max of 98 degrees F

In [186]:
weather_df[['Min_TemperatureF']].min()

Min_TemperatureF    23
dtype: int64

### 7. The coldest day on record got as cold as 23 defrees F

In [187]:
weather_df[['Mean_Visibility_Miles']].mean()

Mean_Visibility_Miles    9.429608
dtype: float64

### 8. The overall avg visibility is 9.42 miles